contracts/Staking.sol

```diff
@@ -157,6 +157,28 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         _decreaseActiveLockAmount(_from, msg.sender, _amount);
     }
 
+    /**
+     * @notice Transfer `_amount` tokens from `_from`'s lock by `msg.sender` to `_to`, and unlock the rest
+     * @param _from Owner of locked tokens
+     * @param _to Recipient of the tokens
+     * @param _amount Number of tokens to be transferred
+     */
+    function unlockAndSlash(
+        address _from,
+        address _to,
+        uint256 _amount
+    )
+        external
+        isInitialized
+    {
+        // No need to check that amount is positive, as _transfer will fail
+        // No need to check that have enough locked funds, as _decreaseActiveLockAmount will fail
+
+        _transfer(_from, msg.sender, _to, address(0), _amount);
+        _decreaseActiveLockAmount(_from, msg.sender, _amount);
+        _unsafeUnlock(_from, msg.sender);
```

Comment:
I haven't analyzed if a reentrance is possible, but it feels safer to do the transfer at the end

Reply:
But there’s no real transfer here, I mean, we are not calling any `token.transfer` (nor any other contract actually).